src/Microsoft.Health.Fhir.Shared.Api/Features/Headers/RawFhirResultExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Globalization;
+using Microsoft.Health.Fhir.Api.Features.ActionResults;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Routing;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.Health.Fhir.Api.Features.Headers
+{
+    public static class RawFhirResultExtensions
+    {
+        public static ResourceWrapperFhirResult SetLocationHeader(this ResourceWrapperFhirResult rawFhirResult, IUrlResolver urlResolver)
```

Comment:
In general I don't really like having both ResourceWrapper and ResourceElement being passed around. ResourceWrapper was supposed to be more of a datalayer concept. I've often gone back and forward on the idea that ResourceElement should have had an interface, but I still feel that it is currently the Domain object for FHIR resource data throughout the pipeline

Reply:
I added an interface and a RawResourceElement class to be passed around instead, let me know if that's closer to what you were thinking.